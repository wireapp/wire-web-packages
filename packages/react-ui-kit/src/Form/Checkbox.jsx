/*
 * Wire
 * Copyright (C) 2017 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {COLOR} from '../index';
import {Input} from './Input';
import PropTypes from 'prop-types';
import React from 'react';
import {Text} from '../Text';
import styled from 'styled-components';

const StyledContainerCheckbox = styled.div`
  /* appearance */
  /* positioning */
  display: flex;
  justify-content: flex-start;
  align-items: center;
`;

const StyledCheckbox = Input.withComponent('input').extend.attrs({
  type: 'checkbox',
})`
  /* appearance */
  /* positioning */
  height: 14px;
  width: 30px;
  margin-bottom: 0;
`;

const Checkbox = ({id, children, style, ...props}) => {
  if (!id) {
    id = Math.random().toString();
  }
  return (
    <StyledContainerCheckbox style={style}>
      <StyledCheckbox id={id} {...props} />
      <label htmlFor={id}>{children}</label>
    </StyledContainerCheckbox>
  );
};

const CheckboxLabel = Text.extend`
  a {
    text-decoration: none;
    color: ${COLOR.LINK};
  }
`;

CheckboxLabel.defaultProps = {
  ...Text.defaultProps,
  bold: true,
  fontSize: '11px',
  textTransform: 'uppercase',
};

Checkbox.propTypes = {
  ...Input.propTypes,
  children: PropTypes.node,
  id: PropTypes.string,
};

Checkbox.defaultProps = {
  ...Input.defaultProps,
  children: null,
  id: null,
};

export {Checkbox, CheckboxLabel};
