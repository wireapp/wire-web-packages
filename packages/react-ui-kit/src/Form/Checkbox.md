Demo:

```js
import {Fragment} from 'react';
import {Column, Columns, Checkbox, CheckboxLabel, Link} from '@wireapp/react-ui-kit';

<Fragment>
  <Columns>
    <Column>Checkbox</Column>
    <Column>
      <Checkbox id="ToU">
        <CheckboxLabel>{'ToU'}</CheckboxLabel>
      </Checkbox>
    </Column>
  </Columns>
  <Columns>
    <Column>Checkbox</Column>
    <Column>
      <Checkbox id="ToULink" defaultChecked={true}>
        <CheckboxLabel>
          {'ToU '}
          <Link href="#">{'component link'}</Link>
          {' text '}
          <a href="#">{'plain link'}</a>
        </CheckboxLabel>
      </Checkbox>
    </Column>
  </Columns>
  <Columns>
    <Column>Disabled Checkbox</Column>
    <Column>
      <Checkbox id="disabled" disabled>
        <CheckboxLabel>{'Disabled'}</CheckboxLabel>
      </Checkbox>
    </Column>
  </Columns>
  <Columns>
    <Column>Invalid Checkbox</Column>
    <Column>
      <Checkbox markInvalid id="ToU">
        <CheckboxLabel>{'ToU'}</CheckboxLabel>
      </Checkbox>
    </Column>
  </Columns>
</Fragment>;
```
