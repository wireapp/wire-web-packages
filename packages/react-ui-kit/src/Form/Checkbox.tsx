/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/** @jsx jsx */
import {jsx} from '@emotion/core';
import {COLOR} from '../Identity';
import {Text, TextProps, textStyles} from '../Text';
import {INPUT_CLASSNAME, Input, InputProps} from './Input';

export interface StyledLabelProps<T = HTMLLabelElement> extends React.HTMLProps<T> {
  disabled?: boolean;
}

const StyledLabel = (props: StyledLabelProps) => {
  const checkSvg =
    '<svg xmlns="http://www.w3.org/2000/svg" width="8" height="6" viewBox="0 0 8 6"><path fill="white" d="M2.8 6L8 .7 7.3 0 2.8 4.6.7 2.4l-.7.7z"/></svg>';
  return (
    <label
      css={{
        [`${INPUT_CLASSNAME}:checked + &::before`]: {
          background: `#000 url('data:image/svg+xml; utf8, ${checkSvg}') no-repeat center`,
        },
        [`${INPUT_CLASSNAME}:focus + &::before`]: {
          borderColor: COLOR.BLUE,
        },
        '&::before': {
          border: '2px solid rgba(0, 0, 0, 0.4)',
          borderRadius: '4px',
          boxSizing: 'border-box',
          content: '""',
          display: 'inline-block',
          height: '16px',
          margin: '0 8px 0 -16px',
          opacity: props.disabled ? 0.56 : 1,
          width: '16px',
        },
        display: 'flex',
        opacity: props.disabled ? 0.56 : 1,
      }}
      {...props}
    />
  );
};

interface CheckboxProps<T = HTMLInputElement> extends InputProps<T> {
  id?: string;
}

const Checkbox: React.SFC<CheckboxProps> = ({id = Math.random().toString(), children, style, disabled, ...props}) => (
  <div
    css={{
      alignItems: 'center',
      display: 'flex',
      justifyContent: 'flex-start',
    }}
    style={style}
  >
    <Input
      type={'checkbox'}
      id={id}
      style={{
        height: '16px',
        marginBottom: '0',
        opacity: 0,
        width: '16px',
      }}
      disabled={disabled}
      {...props}
    />
    <StyledLabel htmlFor={id} disabled={disabled}>
      {children}
    </StyledLabel>
  </div>
);

export interface CheckboxLabelProps<T = HTMLSpanElement> extends TextProps<T> {}

const CheckboxLabel = ({
  bold = true,
  color = COLOR.GRAY_DARKEN_24,
  fontSize = '11px',
  textTransform = 'uppercase',
  ...props
}: CheckboxLabelProps) => (
  <Text
    css={{
      ...textStyles({
        bold,
        color,
        fontSize,
        textTransform,
        ...props,
      }),
      a: {
        color: COLOR.LINK,
        textDecoration: 'none',
      },
    }}
    {...props}
  />
);

export {Checkbox, CheckboxLabel};
