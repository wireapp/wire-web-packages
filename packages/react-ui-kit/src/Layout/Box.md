Demo:

```js
import {Box} from '@wireapp/react-ui-kit';

<Box>
  Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a mattis nibh, sed maximus leo. Fusce a lacinia sem,
  vitae ornare dolor. Quisque rhoncus, magna non lacinia sagittis, erat augue fringilla metus, eu consectetur leo velit
  non lacus. Phasellus ipsum turpis, dapibus ut purus in, lobortis consectetur mi.
</Box>;
```
