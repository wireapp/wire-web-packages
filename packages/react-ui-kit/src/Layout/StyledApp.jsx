import {COLOR} from '../Identity';
import styled from 'styled-components';

const StyledApp = styled.div`
  color: ${COLOR.GRAY_DARKEN_48};
  display: flex;
  flex-direction: column;
  font-family: -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica Neue, Helvetica, Arial, sans-serif;
  font-weight: 300;
  line-height: 1.5;
  min-height: 100vh;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
  * {
    box-sizing: border-box;
  }
`;

export {StyledApp};
