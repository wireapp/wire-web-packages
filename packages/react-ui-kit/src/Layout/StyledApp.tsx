/*
 * Wire
 * Copyright (C) 2018 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

/** @jsx jsx */
import {ObjectInterpolation, jsx} from '@emotion/core';
import {GlobalStyle} from '../GlobalStyle';
import {COLOR} from '../Identity';
import {filterProps} from '../util';

export interface StyledAppContainerProps<T = HTMLDivElement> extends React.HTMLProps<T> {
  backgroundColor?: string;
}

const StyledAppContainerStyles: (props: StyledAppContainerProps) => ObjectInterpolation<undefined> = ({
  backgroundColor = COLOR.GRAY_LIGHTEN_88,
}) => ({
  '*': {
    boxSizing: 'border-box',
  },
  MozOsxFontSmoothing: 'grayscale',
  WebkitFontSmoothing: 'antialiased',
  backgroundColor: backgroundColor,
  color: COLOR.TEXT,
  display: 'flex',
  flexDirection: 'column',
  fontFamily: '-apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica Neue, Helvetica, Arial, sans-serif',
  fontWeight: 300,
  lineHeight: 1.5,
  minHeight: '100vh',
});

const filterStyledAppContainerProps = (props: Object) => filterProps(props, ['backgroundColor']);

const StyledAppContainer = (props: StyledAppContainerProps) => (
  <div css={StyledAppContainerStyles(props)} {...filterStyledAppContainerProps(props)} />
);

const StyledApp = ({children, ...props}) => (
  <StyledAppContainer {...props}>
    <GlobalStyle />
    {children}
  </StyledAppContainer>
);

export {StyledApp};
