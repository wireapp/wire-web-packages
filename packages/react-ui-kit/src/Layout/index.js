import setGlobalStyles from '../globalStyles';

setGlobalStyles();

export * from './Column';
export * from './Container';
export * from './Content';
export * from './Footer';
export * from './Header';
export * from './StyledApp';
