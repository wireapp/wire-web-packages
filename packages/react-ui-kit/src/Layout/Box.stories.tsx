/*
 * Wire
 * Copyright (C) 2025 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import {Meta, StoryObj} from '@storybook/react';

import {Box} from './Box';

import {COLOR} from '../Identity/colors';
import {Text} from '../Text';

const meta: Meta<typeof Box> = {
  component: Box,
  title: 'Layout/Box',
  decorators: [
    Story => (
      <div style={{padding: '24px', maxWidth: '600px', margin: '0 auto'}}>
        <Story />
      </div>
    ),
  ],
};

export default meta;
type Story = StoryObj<typeof Box>;

const LOREM_IPSUM = `Lorem ipsum dolor sit amet, consectetur adipiscing elit. Fusce a mattis nibh, sed maximus leo. Fusce a lacinia sem,
vitae ornare dolor. Quisque rhoncus, magna non lacinia sagittis, erat augue fringilla metus, eu consectetur leo velit
non lacus. Phasellus ipsum turpis, dapibus ut purus in, lobortis consectetur mi.`;

export const Default: Story = {
  args: {
    children: LOREM_IPSUM,
  },
};

export const WithCustomPadding: Story = {
  args: {
    children: LOREM_IPSUM,
    style: {padding: '32px'},
  },
};

export const WithCustomBorder: Story = {
  args: {
    children: LOREM_IPSUM,
    style: {border: `2px solid ${COLOR.BLUE}`},
  },
};

export const WithFormattedContent: Story = {
  args: {
    children: (
      <>
        <Text block style={{fontSize: '20px', fontWeight: 'bold', marginBottom: '16px'}}>
          Important Notice
        </Text>
        <Text block style={{marginBottom: '12px'}}>
          {LOREM_IPSUM}
        </Text>
        <Text block style={{color: COLOR.GRAY}}>
          Last updated: 2025-01-01
        </Text>
      </>
    ),
  },
};

export const Nested: Story = {
  args: {
    children: (
      <>
        <Text block style={{marginBottom: '16px'}}>
          Outer box content
        </Text>
        <Box style={{backgroundColor: COLOR.GRAY_LIGHTEN_92}}>
          <Text>Inner box content</Text>
        </Box>
      </>
    ),
  },
};

export const Interactive: Story = {
  args: {
    children: (
      <div style={{textAlign: 'center'}}>
        <Text block style={{marginBottom: '16px'}}>
          Click the button below
        </Text>
        <button
          style={{
            backgroundColor: COLOR.BLUE,
            border: 'none',
            borderRadius: '4px',
            color: COLOR.WHITE,
            cursor: 'pointer',
            padding: '8px 16px',
          }}
        >
          Action Button
        </button>
      </div>
    ),
    style: {cursor: 'pointer'},
  },
};
