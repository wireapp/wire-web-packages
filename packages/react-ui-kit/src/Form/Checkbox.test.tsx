/*
 * Wire
 * Copyright (C) 2019 Wire Swiss GmbH
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see http://www.gnu.org/licenses/.
 *
 */

import '@testing-library/jest-dom/jest-globals';
import {fireEvent, render} from '@testing-library/react';

import {Checkbox, CheckboxLabel} from './Checkbox';

import {StyledApp} from '../../src/Layout/StyledApp';
import {THEME_ID} from '../Layout';
import {matchComponent} from '../test/testUtil';

/* eslint-disable jest/expect-expect */

describe('"Checkbox"', () => {
  it('renders', () => matchComponent(<Checkbox id="1">Check</Checkbox>));
  it('renders (dark theme)', () => matchComponent(<Checkbox id="1">Checkbox</Checkbox>, THEME_ID.DARK));
  it('renders as invalid', () =>
    matchComponent(
      <Checkbox id="1" markInvalid>
        Check
      </Checkbox>,
    ));
  it('renders disabled', () =>
    matchComponent(
      <Checkbox id="1" disabled>
        Check
      </Checkbox>,
    ));
});

describe('"CheckboxLabel"', () => {
  it('renders', () => matchComponent(<CheckboxLabel>Label</CheckboxLabel>));
});

//TODO: - create custom render for UI-Kit with a styledapp wrapper(SQSERVICES-1672)
test('account creation terms and condition checkbox is checked/unchecked', () => {
  const {getByTestId} = render(
    <StyledApp themeId={THEME_ID.LIGHT}>
      <Checkbox id="1" data-testid="do-terms"></Checkbox>
    </StyledApp>,
  );
  const checkbox = getByTestId('do-terms');
  expect(checkbox).not.toBeChecked();
  fireEvent.click(checkbox);
  expect(checkbox).toBeChecked();
});
