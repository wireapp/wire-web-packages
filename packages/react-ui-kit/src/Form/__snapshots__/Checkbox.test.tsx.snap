// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"Checkbox" renders (dark theme) 1`] = `
.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-3 {
  background: #000;
  -webkit-transition: background 0.15s;
  transition: background 0.15s;
}

.emotion-0 {
  background: rgb(61,61,61);
  border: none;
  border-radius: 4px;
  box-shadow: none;
  caret-color: #2391d3;
  color: #fff;
  font-weight: 300;
  height: 56px;
  line-height: 24px;
  margin: 0 0 16px;
  opacity: 1;
  outline: none;
  padding: 0 16px;
  width: 100%;
}

.emotion-0::-moz-placeholder {
  color: rgb(247,248,250);
  font-size: 11px;
  text-transform: uppercase;
  opacity: 1;
}

.emotion-0::-ms-input-placeholder {
  color: rgb(247,248,250);
  font-size: 11px;
  text-transform: uppercase;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(247,248,250);
  font-size: 11px;
  text-transform: uppercase;
}

.emotion-0:invalid {
  box-shadow: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
}

input:checked + .emotion-1::before {
  background: #000 url('data:image/svg+xml; utf8, <svg xmlns="http://www.w3.org/2000/svg" width="8" height="6" viewBox="0 0 8 6"><path fill="white" d="M2.8 6L8 .7 7.3 0 2.8 4.6.7 2.4l-.7.7z"/></svg>') no-repeat center;
}

input:focus + .emotion-1::before {
  border-color: #2391d3;
}

.emotion-1::before {
  border: 2px solid #fff;
  border-radius: 4px;
  box-sizing: border-box;
  content: "";
  display: inline-block;
  height: 16px;
  margin: 0 8px 0 -16px;
  opacity: 1;
  width: 16px;
}

.emotion-1 a {
  color: #fff;
  display: inline;
  font-size: 11px;
  font-weight: 600;
  opacity: 1;
  text-align: left;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.24s;
  transition: all 0.24s;
}

.emotion-1 a:hover {
  color: rgb(214,214,214);
}

.emotion-1 a:visited,
.emotion-1 a:link,
.emotion-1 a:active {
  color: #fff;
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <input
      className="input emotion-0"
      id="1"
      style={
        Object {
          "height": "16px",
          "marginBottom": "0",
          "opacity": 0,
          "width": "16px",
        }
      }
      type="checkbox"
    />
    <label
      className="emotion-1"
      htmlFor="1"
    >
      Checkbox
    </label>
  </div>
</div>
`;

exports[`"Checkbox" renders 1`] = `
.emotion-3 {
  background: rgb(247,248,250);
  -webkit-transition: background 0.15s;
  transition: background 0.15s;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-0 {
  background: #fff;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  caret-color: #2391d3;
  color: rgb(52,56,59);
  font-weight: 300;
  height: 56px;
  line-height: 24px;
  margin: 0 0 16px;
  opacity: 1;
  outline: none;
  padding: 0 16px;
  width: 100%;
}

.emotion-0::-moz-placeholder {
  color: rgb(141,152,159);
  font-size: 11px;
  text-transform: uppercase;
  opacity: 1;
}

.emotion-0::-ms-input-placeholder {
  color: rgb(141,152,159);
  font-size: 11px;
  text-transform: uppercase;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(141,152,159);
  font-size: 11px;
  text-transform: uppercase;
}

.emotion-0:invalid {
  box-shadow: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
}

input:checked + .emotion-1::before {
  background: #000 url('data:image/svg+xml; utf8, <svg xmlns="http://www.w3.org/2000/svg" width="8" height="6" viewBox="0 0 8 6"><path fill="white" d="M2.8 6L8 .7 7.3 0 2.8 4.6.7 2.4l-.7.7z"/></svg>') no-repeat center;
}

input:focus + .emotion-1::before {
  border-color: #2391d3;
}

.emotion-1::before {
  border: 2px solid rgb(52,56,59);
  border-radius: 4px;
  box-sizing: border-box;
  content: "";
  display: inline-block;
  height: 16px;
  margin: 0 8px 0 -16px;
  opacity: 1;
  width: 16px;
}

.emotion-1 a {
  color: rgb(52,56,59);
  display: inline;
  font-size: 11px;
  font-weight: 600;
  opacity: 1;
  text-align: left;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.24s;
  transition: all 0.24s;
}

.emotion-1 a:hover {
  color: rgb(44,47,50);
}

.emotion-1 a:visited,
.emotion-1 a:link,
.emotion-1 a:active {
  color: rgb(52,56,59);
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <input
      className="input emotion-0"
      id="1"
      style={
        Object {
          "height": "16px",
          "marginBottom": "0",
          "opacity": 0,
          "width": "16px",
        }
      }
      type="checkbox"
    />
    <label
      className="emotion-1"
      htmlFor="1"
    >
      Check
    </label>
  </div>
</div>
`;

exports[`"Checkbox" renders as invalid 1`] = `
.emotion-3 {
  background: rgb(247,248,250);
  -webkit-transition: background 0.15s;
  transition: background 0.15s;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-0 {
  background: #fff;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  caret-color: #2391d3;
  color: rgb(52,56,59);
  font-weight: 300;
  height: 56px;
  line-height: 24px;
  margin: 0 0 16px;
  opacity: 1;
  outline: none;
  padding: 0 16px;
  width: 100%;
}

.emotion-0::-moz-placeholder {
  color: rgb(141,152,159);
  font-size: 11px;
  text-transform: uppercase;
  opacity: 1;
}

.emotion-0::-ms-input-placeholder {
  color: rgb(141,152,159);
  font-size: 11px;
  text-transform: uppercase;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(141,152,159);
  font-size: 11px;
  text-transform: uppercase;
}

.emotion-0:invalid {
  box-shadow: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 1;
}

input:checked + .emotion-1::before {
  background: #000 url('data:image/svg+xml; utf8, <svg xmlns="http://www.w3.org/2000/svg" width="8" height="6" viewBox="0 0 8 6"><path fill="white" d="M2.8 6L8 .7 7.3 0 2.8 4.6.7 2.4l-.7.7z"/></svg>') no-repeat center;
}

input:focus + .emotion-1::before {
  border-color: #2391d3;
}

.emotion-1::before {
  border: 2px solid #fb0807;
  border-radius: 4px;
  box-sizing: border-box;
  content: "";
  display: inline-block;
  height: 16px;
  margin: 0 8px 0 -16px;
  opacity: 1;
  width: 16px;
}

.emotion-1 a {
  color: rgb(52,56,59);
  display: inline;
  font-size: 11px;
  font-weight: 600;
  opacity: 1;
  text-align: left;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.24s;
  transition: all 0.24s;
}

.emotion-1 a:hover {
  color: rgb(44,47,50);
}

.emotion-1 a:visited,
.emotion-1 a:link,
.emotion-1 a:active {
  color: rgb(52,56,59);
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <input
      className="input emotion-0"
      id="1"
      style={
        Object {
          "height": "16px",
          "marginBottom": "0",
          "opacity": 0,
          "width": "16px",
        }
      }
      type="checkbox"
    />
    <label
      className="emotion-1"
      htmlFor="1"
    >
      Check
    </label>
  </div>
</div>
`;

exports[`"Checkbox" renders disabled 1`] = `
.emotion-3 {
  background: rgb(247,248,250);
  -webkit-transition: background 0.15s;
  transition: background 0.15s;
}

.emotion-2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.emotion-0 {
  background: #fff;
  border: none;
  border-radius: 4px;
  box-shadow: none;
  caret-color: #2391d3;
  color: rgb(52,56,59);
  font-weight: 300;
  height: 56px;
  line-height: 24px;
  margin: 0 0 16px;
  opacity: 0.56;
  outline: none;
  padding: 0 16px;
  width: 100%;
}

.emotion-0::-moz-placeholder {
  color: rgb(141,152,159);
  font-size: 11px;
  text-transform: uppercase;
  opacity: 1;
}

.emotion-0::-ms-input-placeholder {
  color: rgb(141,152,159);
  font-size: 11px;
  text-transform: uppercase;
}

.emotion-0::-webkit-input-placeholder {
  color: rgb(141,152,159);
  font-size: 11px;
  text-transform: uppercase;
}

.emotion-0:invalid {
  box-shadow: none;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  opacity: 0.56;
}

input:checked + .emotion-1::before {
  background: #000 url('data:image/svg+xml; utf8, <svg xmlns="http://www.w3.org/2000/svg" width="8" height="6" viewBox="0 0 8 6"><path fill="white" d="M2.8 6L8 .7 7.3 0 2.8 4.6.7 2.4l-.7.7z"/></svg>') no-repeat center;
}

input:focus + .emotion-1::before {
  border-color: #2391d3;
}

.emotion-1::before {
  border: 2px solid rgb(52,56,59);
  border-radius: 4px;
  box-sizing: border-box;
  content: "";
  display: inline-block;
  height: 16px;
  margin: 0 8px 0 -16px;
  opacity: 0.56;
  width: 16px;
}

.emotion-1 a {
  color: rgb(52,56,59);
  display: inline;
  font-size: 11px;
  font-weight: 600;
  opacity: 1;
  text-align: left;
  text-transform: uppercase;
  cursor: pointer;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 0.24s;
  transition: all 0.24s;
}

.emotion-1 a:hover {
  color: rgb(44,47,50);
}

.emotion-1 a:visited,
.emotion-1 a:link,
.emotion-1 a:active {
  color: rgb(52,56,59);
}

<div
  className="emotion-3"
>
  <div
    className="emotion-2"
  >
    <input
      className="input emotion-0"
      disabled={true}
      id="1"
      style={
        Object {
          "height": "16px",
          "marginBottom": "0",
          "opacity": 0,
          "width": "16px",
        }
      }
      type="checkbox"
    />
    <label
      className="emotion-1"
      disabled={true}
      htmlFor="1"
    >
      Check
    </label>
  </div>
</div>
`;

exports[`"CheckboxLabel" renders 1`] = `
.emotion-1 {
  background: rgb(247,248,250);
  -webkit-transition: background 0.15s;
  transition: background 0.15s;
}

.emotion-0 {
  color: rgb(52,56,59);
  display: inline;
  font-size: 16px;
  font-weight: 300;
  opacity: 1;
  text-align: left;
  text-transform: none;
  color: rgb(141,152,159);
  display: inline;
  font-size: 11px;
  font-weight: 600;
  opacity: 1;
  text-align: left;
  text-transform: uppercase;
}

.emotion-0 a {
  color: rgb(52,56,59);
  -webkit-text-decoration: none;
  text-decoration: none;
}

<div
  className="emotion-1"
>
  <span
    className="emotion-0"
  >
    Label
  </span>
</div>
`;
